drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -395,7 +399,16 @@ int stm32_i2c_msg_write(struct device *dev, struct i2c_msg *msg,
 	stm32_i2c_enable_transfer_interrupts(dev);
 	LL_I2C_EnableIT_TX(i2c);
 
-	k_sem_take(&data->device_sync_sem, K_FOREVER);
+	err = k_sem_take(&data->device_sync_sem,
+			 cfg->timeout == K_FOREVER ? K_FOREVER :
+						     K_MSEC(cfg->timeout));
```

Comment:
Any way to do this calculation only at init and store it once for all?

Reply:
Done it at compile time.

Reply:
Good!